Extensions/DraggableBehavior/tests/draggableruntimebehavior.spec.js

```diff
@@ -51,112 +51,211 @@ describe('gdjs.DraggableRuntimeBehavior', function () {
-
-    expect(object.getX()).to.be(750);
-    expect(object.getY()).to.be(700);
-    expect(object2.getX()).to.be(100);
-    expect(object2.getY()).to.be(200);
+  describe('(touch)', function () {
+    it('can drag an object', function () {
+      runtimeGame.getInputManager().touchSimulateMouse(false);
+      object.setPosition(450, 500);
+
+      // Drag'n'drop
+      runtimeScene.renderAndStep(1000 / 60);
+      runtimeGame.getInputManager().onTouchStart(1, 10, 20);
+      runtimeGame.getInputManager().onTouchStart(0, 450, 500);
+      runtimeScene.renderAndStep(1000 / 60);
+      runtimeGame.getInputManager().onFrameEnded();
+      runtimeGame.getInputManager().onTouchMove(0, 750, 600);
+      runtimeScene.renderAndStep(1000 / 60);
+      runtimeGame.getInputManager().onFrameEnded();
+      runtimeGame.getInputManager().onTouchEnd(0);
+      runtimeGame.getInputManager().onTouchEnd(1);
+      runtimeScene.renderAndStep(1000 / 60);
+      runtimeGame.getInputManager().onFrameEnded();
+
+      expect(object.getX()).to.be(750);
+      expect(object.getY()).to.be(600);
+
+      // Move another unrelated touch
+      runtimeGame.getInputManager().onTouchMove(1, 750, 600);
+      runtimeScene.renderAndStep(1000 / 60);
+      runtimeGame.getInputManager().onTouchMove(1, 850, 700);
+      runtimeScene.renderAndStep(1000 / 60);
+
+      expect(object.getX()).to.be(750);
+      expect(object.getY()).to.be(600);
+
+      // Start drag'n'drop with another touch
+      runtimeGame.getInputManager().onTouchEnd(1);
+      runtimeGame.getInputManager().onFrameEnded();
+      runtimeScene.renderAndStep(1000 / 60);
+      runtimeGame.getInputManager().onTouchStart(1, 750, 600);
+      runtimeScene.renderAndStep(1000 / 60);
+      runtimeGame.getInputManager().onTouchMove(1, 850, 700);
+      runtimeScene.renderAndStep(1000 / 60);
+      runtimeGame.getInputManager().onTouchEnd(1);
+      runtimeGame.getInputManager().onFrameEnded();
+
+      expect(object.getX()).to.be(850);
+      expect(object.getY()).to.be(700);
+    });
+    it('can drag 2 objects with multitouch', function () {
+      runtimeGame.getInputManager().touchSimulateMouse(false);
+      object.setPosition(450, 500);
+      object2.setPosition(650, 600);
+
+      // Drag'n'drop
+      runtimeScene.renderAndStep(1000 / 60);
+      runtimeGame.getInputManager().onTouchStart(2, 450, 500);
+      runtimeGame.getInputManager().onTouchStart(1, 650, 600);
+      runtimeScene.renderAndStep(1000 / 60);
+      runtimeGame.getInputManager().onFrameEnded();
+      runtimeGame.getInputManager().onTouchMove(2, 750, 700);
+      runtimeGame.getInputManager().onTouchMove(1, 100, 200);
+      runtimeScene.renderAndStep(1000 / 60);
+      runtimeGame.getInputManager().onFrameEnded();
+      runtimeGame.getInputManager().onTouchEnd(1);
+      runtimeGame.getInputManager().onTouchEnd(2);
+      runtimeScene.renderAndStep(1000 / 60);
+      runtimeGame.getInputManager().onFrameEnded();
```

Comment:
Is this extra frame required?

Reply:
I moved it after the checks. Without it the following test handle this event too and move the object.